packages/headless/src/features/attached-results/attached-results-slice.ts

```diff
@@ -0,0 +1,40 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {
+  setAttachedResults,
+  attachResult,
+  detachResult,
+} from './attached-results-actions';
+import {getAttachedResultsInitialState} from './attached-results-state';
+
+export const attachToCaseReducer = createReducer(
+  getAttachedResultsInitialState(),
+
+  (builder) => {
+    builder
+      .addCase(setAttachedResults, (state, action) => {
+        const {results: attachedResults, message} = action.payload;
+
+        if ('attachedResults' in state && state['results']?.length > 0) {
+          return;
+        }
+
+        state['results'] = attachedResults;
+        state['message'] = message;
+      })
+      .addCase(attachResult, (state, action) => {
+        state.results = [action.payload.result, ...state.results];
+      })
+      .addCase(detachResult, (state, action) => {
+        state.results = state.results.filter((result) => {
+          const isPermanentIdEqual =
+            result?.permanentId !== undefined &&
+            result?.permanentId === action.payload?.result?.permanentId;
+          const isUriHashEqual =
+            result?.uriHash !== undefined &&
+            result?.uriHash === action.payload?.result?.uriHash;
+          const isCaseIdEqual = result.caseId === action.payload.result.caseId;
+          return !isCaseIdEqual || (!isPermanentIdEqual && !isUriHashEqual);
```

Comment:
I'm not sure I understand where the attached results are originally gonna be coming from. Could we assume that, like the Search API, it has a `uniqueId`?

Also, I think this logic should be extracted into a `attachedResultsEqual` function.

Reply:
The attached results are coming from a record in a CRM (for example Salesforce)
And the goal is to compare them sometimes with results coming from the SearchAPI.
It's confusing but these are two different structures :(